dispatchers.go

```diff
@@ -234,7 +234,9 @@ func from(b *Builder, args []string, attributes map[string]bool, flagArgs []stri
 			return fmt.Errorf("Windows does not support FROM scratch")
 		}
 	}
+	defaultArgs := envMapAsSlice(builtinBuildArgs)
```

Comment:
Are the default values all correct when `--platform` is specified differently for different stages?

Reply:
@nalind mostly we only use `$BUILDPLATFORM` or `$TARGETPLATFORM` in inline platform syntax so yes default values are correct for different stages. However there is still a small issue with the way buildah populates `$TARGETPLATFORM` but its a small fix on buildah side not on imagebuilder side.

Reply:
Created a PR in buildah which exercises this and fixes `$TARGETPLATFORM` issue: https://github.com/containers/buildah/pull/3899

Reply:
What about the `dockerclient` client?

Reply:
The behavior not being in `docker build` means that we can't write a conformance test for it, and that is very unfortunate, but that doesn't tell me whether someone who uses the `imagebuilder` CLI will get a surprising result if their Dockerfile tries to reference these arguments.

A different change we previously made at the parsing level for `buildah`'s sake caused `imagebuilder` to switch from reporting an error when a specific option was used to silently ignoring it, and that tripped up our colleagues who use `imagebuilder` when they found it no longer complained when they tried to use the option.  IIRC they got pretty far down the road before noticing it, since ignoring that option still produced the correct result some of the time.  I'm trying to avoid repeating that mistake.

Reply:
@nalind hmm that is true do you have any suggestion about what should we do for `dockerclient` for such cases ? Would a warning work here.

Reply:
It would need to either flag the attempt to use the unsupported argument as an error, or expand it correctly.  What happens with the current patch?

Reply:
> @nalind For `dockerclient` we are already setting args correctly so `$BUILDPLATFORM` will be set correctly but other options like `$TARGETPLATFORM` will be only exercised by docker deamon if `"features": { "buildkit": true }` is enabled on the server side. So if docker daemon is running without buildkit certain things will be `no-op` but I think imagebuilder can not help a lot in such cases.

Its pretty much like above everything works fine except its just that `$TARGETPLATFORM` might be `no-op` depending upon the fact if `dockerserver` supports buildkit or not but I also think this part might be out of the scope of this PR itself.

But PR only talks about implementation of `FROM --platform=` which works fine for `dockerclient` and we added implementation for dockerclient here: https://github.com/openshift/imagebuilder/pull/212

Reply:
The docker daemon doesn't evaluate the Dockerfile, we do.

Reply:
> The docker daemon doesn't evaluate the Dockerfile, we do.

Oh my bad i mixed things, if we use `TARGETPLATFORM` as `-build-arg` with `imagebuilder` it will work but we can't expect `$TARGETPLATFORM` to get correctly populated cause we don't have cli arg like `buildah build --platform` or `docker build --platform` in imagebuilder to correctly set the value of `TARGETPLATFORM` correct me if i got this wrong.

So its just that with `imagebuilder` it is expected from user to set the value of `TARGETGETPLATFORM` using `-build-arg` otherwise there is no way set the correct value of `TARGETPLATFORM` and default value is always https://github.com/openshift/imagebuilder/blob/master/dispatchers.go#L35

I have a feeling that this is just expected behavior and I'm overthinking this.

Reply:
Wouldn't TAGETGETPLATFORM just default to the local platform then?